docs/boot.md

```diff
@@ -0,0 +1,48 @@
+## There is a more detailed explanation in the [circuit python docs](https://docs.circuitpython.org/en/latest/README.html), however there are some common use cases for your keyboard listed here
+
+### You can hide your device from showing up as a usb storage
+
+```python
+storage.disable_usb_drive()
+```
+
+### Make your keyboard work in bios
+
+```python
+usb_hid.enable(boot_device=1)
+```
+
+### Disable serial comms
+
+```python
+# Equivalent to usb_cdc.enable(console=False, data=False)
+usb_cdc.disable()
+```
+
+### A fully working example, with a
+
+```python
+import supervisor
+import board
+import digitalio
+import storage
+import usb_cdc
+import usb_hid
+
+supervisor.set_next_stack_limit(4096 + 4096)
+
+# If this key is held during boot, don't run the code which hides the storage and disables serial
```

Comment:
Add something along the lines: The exact pins are dependend on your hardware wiring.

Reply:
@tannewt provided an example of his boot.py in his [repo](https://github.com/tannewt/keebloop/blob/main/boot.py).  He doesn't use KMK but he loads his board configuration in boot.py and uses it to detect if there are any keys pressed while booting.  If so,  then the different USB devices are not disabled.

I suspect a similar approach could be used if kb.py is loaded.

For the PyKey series of boards, they all have the same row/column pin names as part of the boards and it's easy to hardcode the pin names to first row/first column.  See [here](https://github.com/jpconstantineau/pykey/blob/main/pykey_boards_production/pykey60/boot.py) for what I use.  I suspect it's almost identical.

However, for any other boards, we would need to be able to pick up dynamically what pins are used.

As a first pass, this example is good but comments on what needs to be changed would be useful. Especially why the output/input config and why it's in a specific way. (diodes are pointing which way...)

Reply:
I wonder if you do have compile issues in kb.py if you really get stuck or if it simply bails and goes into safe mode with the USB drive enabled with the error messages in boot_out.txt.  Even if it doesn't go in safe mode, wouldn't it not run the USB disabling code that prevents you to go in programming mode?

Give me some code I can try and I can see what really happens.  If my boards do get locked out, I do have a jlink to erase and reflash everything from the bootloader.  

The only way I could potentially see why you would get locked out is in cases where your GPIOs are valid (not a compile error) but not connected to switches or set with the wrong diode direction.  It would never be able to detect when you press the switch and thus always automatically disable USB.  Perhaps, the documentation should refer to the method of how getting in safe mode for those trying to configure their boot.py but incorrectly do so and lock themselves out.

Reply:
That is probably right. Let's take the discussion from this PR into an issue.